File: src/main/java/xyz/emirdev/echogen/events/PlayerJoinListener.java
package xyz.emirdev.echogen.events;

import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import xyz.emirdev.echogen.Echogen;
import xyz.emirdev.echogen.task.RamBarTask;
import xyz.emirdev.echogen.task.TPSBarTask;
import xyz.emirdev.echogen.utils.Utils;

public class PlayerJoinListener implements Listener {

    @EventHandler()
    public void onJoin(PlayerJoinEvent event) {
        if (RamBarTask.instance().hasPlayer(event.getPlayer().getUniqueId())) {
            RamBarTask.instance().addPlayer(event.getPlayer());
        }
        if (TPSBarTask.instance().hasPlayer(event.getPlayer().getUniqueId())) {
            TPSBarTask.instance().addPlayer(event.getPlayer());
        }

        String joinMessage = Echogen.get().getPluginConfig().getRootNode().node("chat", "messages", "join").getString();
        if (joinMessage == null) return;

        if (joinMessage.isEmpty()) {
            event.joinMessage(null);
            return;
        }

        Player player = event.getPlayer();

        Utils.formatMessage(joinMessage,
                Placeholder.parsed("name", player.getName())
        );
    }
}

File: src/main/kotlin/cat/emir/echogen/events/PlayerJoinListener.kt
package cat.emir.echogen.events

import me.clip.placeholderapi.PlaceholderAPI
import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import cat.emir.echogen.Echogen
import cat.emir.echogen.task.RamBarTask
import cat.emir.echogen.task.TPSBarTask
import cat.emir.echogen.toComponent

class PlayerJoinListener(val plugin: Echogen) : Listener {

    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        val player = event.player

        listOf(RamBarTask.instance(plugin), TPSBarTask.instance(plugin)).forEach {
            if (it.hasPlayer(player.uniqueId))
                it.addPlayer(player)
        }

        var joinMessage = plugin.config.rootNode.node("chat", "messages", "join").string ?: return

        if (joinMessage.isEmpty()) {
            event.joinMessage(null)
            return
        }

        if (plugin.isPAPIEnabled)
            joinMessage = PlaceholderAPI.setPlaceholders(player, joinMessage)

        event.joinMessage(joinMessage.toComponent(
            Placeholder.parsed("name", player.name)
        ))
    }
}
